SSA/Experimental/IntrinsicAsymptotics.lean

```diff
@@ -1450,17 +1513,31 @@ def ex1_lhs : ICom ExOp [.nat] .nat :=
 def ex1_rhs : ICom ExOp [.nat] .nat :=
   ICom.ret ⟨0, by simp[Ctxt.snoc]⟩
 
-def p1 : PeepholeRewrite ExOp [.nat] .nat:=
-  { lhs := ex1_lhs, rhs := ex1_rhs, correct := by
-      rw [ex1_lhs, ex1_rhs]
+/-- running `f(x) = <whatever>` 0 times is the identity. -/
+def ex1'_lhs : ICom ExOp [.nat] .nat :=
+  ICom.lete (rgn (k := 0) ⟨0, by simp[Ctxt.snoc]⟩ (
+      Reg.mvar 0)
+  ) <|
+  ICom.ret ⟨0, by simp[Ctxt.snoc]⟩
+
+def ex1'_rhs : ICom ExOp [.nat] .nat :=
+  ICom.ret ⟨0, by simp[Ctxt.snoc]⟩
+
+def p1' : PeepholeRewrite ExOp [.nat] .nat:=
+  { lhs := ex1'_lhs, rhs := ex1'_rhs, correct := by
+      rw [ex1'_lhs, ex1'_rhs]
       simp_peephole [add, rgn]
-      simp
+      intros a
+    -- | @chris: what's the correct lemma we need
+    -- to automate this proof?
+      simp[Ctxt.Valuation.snoc]
```

Comment:
What's the correct `simp` lemma instead of this? The proof state is:

```lean
Ctxt.Valuation.snoc ll✝ ((fun val => mv✝ [ExTy.nat] ExTy.nat 0 fun _ty _var => val)^[0] a)
    (Var.last [ExTy.nat] ExTy.nat) =
  a
```